File: backend/order/src/main/java/emsi/moncef/order/models/Order.java
package emsi.moncef.order.models;

import emsi.moncef.order.enums.Status;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "order_tbl") //Order est un mot clé réservé en SQL
@Builder
@Getter
@Setter
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false, unique = true)
    private String orderNumber;
    private LocalDateTime orderDate;
    private String customerName;
    private String customerEmail;
    private String shippingAddress;
    @Enumerated(EnumType.STRING)
    private Status status;
    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<OrderLine> orderLines = new ArrayList<>();
}

File: frontend/types/order.ts
export interface OrderLine {
  skuCode: string
  quantity: number
  orderId: number
}

export interface Order {
  orderNumber: string
  orderDate: string
  customerName: string
  customerEmail: string
  shippingAddress: string
  status: 'PENDING' | 'CONFIRMED' | 'SHIPPED' | 'DELIVERED' | 'REJECTED'
  orderLines: OrderLine[]
}

